lib/install/dnsmasq.rb

```diff
@@ -0,0 +1,85 @@
+require "thor"
+require_relative "../doctor/dnsmasq"
+
+module Install
+  class Dnsmasq
+    def call
+      check_macos
+      install_dnsmasq
+      configure_etc_resolver_devgovuk
+      configure_usr_local_etc_dnsmasq
+      configure_usr_local_etc_dnsmasq_developmentconf
+      restart_dnsmasq
+      puts "✅ Dnsmasq installation and configuration complete!"
+    end
+
+  private
+
+    def macos?
+      %x[uname -a].include?("Darwin")
+    end
+
+    def check_macos
+      return if macos?
+
+      puts "This script currently only works on MacOS."
+      puts "Are you sure you want to continue?"
+      Thor::Shell::Basic.new.yes?
+    end
+
+    def install_dnsmasq
+      return if Doctor::Dnsmasq.new.installed?
+
+      puts "⏳ Installing dnsmasq"
+      system("brew install dnsmasq")
+    end
+
+    def configure_etc_resolver_devgovuk
+      write_file(
```

Comment:
Is this going to destroy the current contents of the file if they exist?
Do we want to warn people of that ?

Reply:
That's a good idea, I've changed the whole script to require confirmation before continuing and it outputs to the user what it's going to do.